### ISSUE TITLE
Pass arguments to call function of custom models

### ISSUE THREAD

**Issue by skeydan** [rstudio/keras#497]
When using custom models that have certain layers (e.g., dropout, batchnorm), we often need to be able to pass a ```training_phase``` argument to the respective layers.
For example:

```
model <-
  function(name = NULL) {
    keras_model_custom(name = name, function(self) {
      self$fc1 <- layer_dense(units = 7 * 7 * 64, use_bias = FALSE)
      self$batchnorm1 <- layer_batch_normalization()
      self$relu1 <- layer_activation_relu()
      
      function(inputs,
               mask = NULL,
               training = TRUE) {
        self$fc1(inputs) %>%
          self$batchnorm1(training = training) %>%
          self$relu1() 
      }
    })
  }
```
As far as I see this is not possible with the current implementation (if I'm not getting it wrong :-))

**Comment by jjallaire** [comment]
This should fix it: https://github.com/rstudio/keras/commit/82af26e73d12c6051d0edf29338e30b72e8091ae

Let me know if that works as expected on your end.

**Comment by skeydan** [comment]
Thanks for the quick implementation!
However, either I'm doing something wrong or ... 

If I modify the example from the docs

```
keras_model_simple_mlp <- function(num_classes, 
                                   use_bn = FALSE, use_dp = FALSE, 
                                   name = NULL) {
  
  # define and return a custom model
  keras_model_custom(name = name, function(self) {
    
    # create layers we'll need for the call (this code executes once)
    self$dense1 <- layer_dense(units = 32, activation = "relu")
    self$dense2 <- layer_dense(units = num_classes, activation = "softmax")
    if (use_dp)
      self$dp <- layer_dropout(rate = 0.5)
    if (use_bn)
      self$bn <- layer_batch_normalization(axis = -1)
    
    # implement call (this code executes during training & inference)
    function(inputs, mask = NULL, training = TRUE) {
      x <- self$dense1(inputs)
      if (use_dp)
        x <- self$dp(x)
      if (use_bn)
        x <- self$bn(x)
      self$dense2(x)
    }
  })
}

model <- keras_model_simple_mlp(num_classes = 10, use_dp = TRUE)

# Generate dummy data
data <- tf$random_normal(c(1000L,100L))
model(data)
```

I cannot do

```
model(inputs, training = FALSE)
```

but strangely, I cannot do any of these either

```
model(data, mask = NULL)
model(inputs = data)
```

The one and only call that works is the one with one argument, no keyword.

I should probably mention it all applies to eager execution turned on...

**Comment by jjallaire** [comment]
Okay, I just revered and will take another look!
